### PR TITLE
Add rounded corners to UI components

### LINKED ISSUE #5811: Add rounded corners in LMS UI components
Integrate rounded corners from [frontend-shared](https://github.com/hypothesis/frontend-shared/pull/1289), like it was done [in client](https://github.com/hypothesis/client/issues/5602).

### PR DESCRIPTION
**Pull request by acelaya**
Take advantage of rounded corner capabilities from [frontend-shared](https://github.com/hypothesis/frontend-shared/pull/1289), and add missing rounded corners where needed for consistency.

The visual changes are relatively small, so I'm adding this with no feature flags.

> **Note**
> This PR is easier to review ignoring whitespaces https://github.com/hypothesis/lms/pull/5815/files?w=1

### Testing steps

Sanity check an assignment, including:

- Grading bar, with and without comment enabled.
- Content picker (edit and create)
- Different content types.
- Errors (authorization and such).

Closes #5811

### CODE REVIEW

**Comment by robertknight** [comment]
I'm seeing that the bottom of the content type form is not rounded. Otherwise this looks good:

Content type form bottom

**Comment by acelaya** [comment]
Since this PR introduces visual changes without a feature flag to control them, I won't be merging it yet, until we have enabled the feature in client.

In fact, we should merge and deploy this more or less at the same time we enable rounded corners in client for LMS users.

File: lms/static/scripts/frontend_apps/components/FilePickerApp.tsx [hunk 1]
```diff
@@ -345,7 +345,7 @@ export default function FilePickerApp({ onSubmit }: FilePickerAppProps) {
           </RouterLink>
         )}
         {/* Card constrains overflow-scroll children to height constraints */}
-        <Card classes="flex flex-col min-h-0">
+        <Card classes="flex flex-col min-h-0 overflow-hidden">
```
Comment (acelaya) [inline lms/static/scripts/frontend_apps/components/FilePickerApp.tsx:348]: Temporary solution until something better is in place for `Cards` in general.

**Review by robertknight — APPROVED** [review]
LGTM

**Pull request merged**